GitUI/BranchTreePanel/SubmoduleNode.cs

```diff
@@ -101,7 +99,14 @@ internal override void OnSelected()
 
         internal override void OnDoubleClick()
         {
-            Open();
+            if (!IsCurrent)
+            {
+                Open();
+            }
+            else
+            {
+                LaunchGitExtensions();
+            }
```

Comment:
I prefer positive logic although the first branch handles the special case then:
```suggestion
            if (IsCurrent)
            {
                LaunchGitExtensions();
            }
            else
            {
                Open();
            }
```

Reply:
ok

Reply:
added a comment too
Note that the approved #10146 depends on this PR, can it be approved?